src/mui-themeable.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import DefaultRawTheme from './styles/raw-themes/light-raw-theme';
+import ThemeManager from './styles/theme-manager';
+
+function getDisplayName(WrappedComponent) {
+  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
+}
+
+export default function muiThemeable(WrappedComponent) {
+  const MuiComponent = (props, {muiTheme = ThemeManager.getMuiTheme(DefaultRawTheme)}) => {
+    return <WrappedComponent {...props} muiTheme={muiTheme} />;
+  };
+
+  MuiComponent.displayName = `mui(${getDisplayName(WrappedComponent)})`;
```

Comment:
Mm, just to play devil's advocate, for this particular use case, could it make sense to just clone any element that is passed, and pass a prop that dictates that it belongs to the `app-bar`, and it's up to the icon-element (whether it's a `IconMenu`, `IconButton`, or `FlatButton` to dictate how style change given that it is in an `app-bar`? 

That way if there are new types of components that support being a `icon-element`, we don't have to add them to the switch condition `AppBar`.